hudi-hadoop-common/src/main/java/org/apache/hudi/storage/hadoop/HoodieHadoopStorage.java

```diff
@@ -202,8 +204,8 @@ public Object getFileSystem() {
   }
 
   @Override
-  public Object getConf() {
-    return fs.getConf();
+  public StorageConfiguration<Configuration> getConf() {
+    return new HadoopStorageConfiguration(fs.getConf());
   }
```

Comment:
Can we add another short-cut method like `getHadoopConf` and make the return type as Hadoop `Configuration` so we can eliminate those explicit type cast from the callers. Or it's better if we can make the return type generic.

Reply:
The idea is that the `StorageConfiguration` object will be passed around (see #11071).  Right now there's still mixed usage of `FileSystem` vs `HoodieStorage` so there needs to be adaptation code and class cast.  Eventually, there is no need to use `Configuration` thus I did not add `getHadoopConf`.

Reply:
As discussed, added `HoodieStorage#unwrapConf` as the shortcut.